### DISCUSSION TITLE
attu 2.2.8 failed to show collections

### DISCUSSION THREAD

**Discussion by Huangpingshan** [milvus-io/milvus#30811] [Q&A and General discussion]
attu 2.2.8 failed to show collections
Now my Milvus version is 2.2.9, when I use attu 2.2.8 connect to Milvus service, it can't show collections.
![image](https://github.com/milvus-io/milvus/assets/84720705/ace439ae-2f36-48bd-872f-b5b3b95df23e)
From Proxy log, we can see
![image](https://github.com/milvus-io/milvus/assets/84720705/80f79242-ecb2-40db-9a45-4d454b6f0078)
And by watching source code, can find the warning is generated by the following code snippet (internal/proxy/task.go)
![image](https://github.com/milvus-io/milvus/assets/84720705/3f6119a0-5638-494d-8391-988b15a48ee5)
The map 'IDs2Names' dose not have required ID,
What should i do to resolve this problem?
And I think the log here should print 'id', not 'collectionName'
![image](https://github.com/milvus-io/milvus/assets/84720705/c273881b-238a-47b1-9739-4353332a3315)

**Comment by yhmo** [comment]
I just use milvus v2.2.9 and attu v2.2.8 to test, it works well.
It seems the log shows "GetPreviledgeExeObj err", I guess maybe it is an error of incorrect usage of RBAC.

**Reply by Huangpingshan** [reply]
Yes, it was always fine before. This situation has only occurred recently.  
What should I do to resolve "GetPreviledgeExeObj err"?

**Reply by yhmo** [reply]
The same issue: https://github.com/milvus-io/milvus/issues/27258
So far the root cause is unknown.

**Comment by yhmo** [comment]
Use python sdk to test instead of Attu.

```
connections.connect(host='localhost', port='19530', user="root", password="Milvus")
print(utility.get_server_version())

collections = utility.list_collections()
print(f"Number of collections: {len(collections)}")
for name in collections:
    collection = Collection(name)
    print(collection.describe())
```

**Reply by Huangpingshan** [reply]
Use python sdk is success to list collections.  
From source code, I think the reason is when use python sdk , code will enter 'else' branch
![image](https://github.com/milvus-io/milvus/assets/84720705/2b8d0c88-5f1e-4a7d-87cc-4244c334d07a)
but when use attu , code will enter 'if sct.GetType() == milvuspb.ShowType_InMemory'  and warning occurd
![image](https://github.com/milvus-io/milvus/assets/84720705/0d2db7d4-6f11-48e5-aae0-7c3b11b41216)
![image](https://github.com/milvus-io/milvus/assets/84720705/5c25dae1-5d9b-4a6a-80b8-1386fd01daef)

**so, is this a bug?**

**Reply by Huangpingshan** [reply]
Watching Milvus 2.3.3 source code,  from 2.2.9 to 2.3.3 , the source code has been changed. Now, the source code is (internal/proxy/task.go):
![image](https://github.com/milvus-io/milvus/assets/84720705/eeabe64a-8a1b-4f2d-9f8b-f4d1950254c0)
Do you hava any idea?

**Reply by yhmo** [reply]
In the pymilvus, the ShowType is set to 0. I am not sure what value is assigned in the Node sdk, let me ask the maintainer.

**Reply by yhmo** [reply]
The Node sdk default sets ShowType to 0. 
We don't know why your attu gets this error. We could not reproduce this error on our machine. If you can reproduce it with a fresh milvus and fresh attu, you can create an issue for us.
